packages/react-instantsearch-hooks-dom/src/widgets/__tests__/Highlight.test.tsx

```diff
@@ -0,0 +1,283 @@
+          </span>
+          <span
+            class="ais-Highlight-nonHighlighted"
+          >
+            nothing
+          </span>
+        </span>
+      </div>
+    `);
+  });
+
+  test('renders path to match', () => {
+    const { container } = render(
+      <Highlight
+        hit={{
+          objectID: '1',
+          __position: 1,
+          data: { subdata: 'test' },
+          _highlightResult: {
+            data: {
+              subdata: {
+                matchedWords: ['test'],
+                matchLevel: 'partial',
+                value: '<mark>te</mark>st',
+              },
+            },
+          },
+        }}
+        // @ts-expect-error TypeScript type does not accept string, as a way to discourage it
+        attribute="data.subdata"
+      />
+    );
+
+    expect(container).toMatchInlineSnapshot(`
+      <div>
+        <span
+          class="ais-Highlight"
+        >
+          <mark
+            class="ais-Highlight-highlighted"
+          >
+            te
+          </mark>
+          <span
+            class="ais-Highlight-nonHighlighted"
+          >
+            st
+          </span>
+        </span>
+      </div>
+    `);
+  });
+
+  test('renders path to match as array', () => {
+    const { container } = render(
+      <Highlight
+        hit={{
+          objectID: '1',
+          __position: 1,
+          data: { subdata: 'test' },
+          _highlightResult: {
+            data: {
+              subdata: {
+                matchedWords: ['test'],
+                matchLevel: 'partial',
+                value: '<mark>te</mark>st',
+              },
+            },
+          },
+        }}
+        attribute={['data', 'subdata']}
+      />
+    );
+
+    expect(container).toMatchInlineSnapshot(`
+      <div>
+        <span
+          class="ais-Highlight"
+        >
+          <mark
+            class="ais-Highlight-highlighted"
+          >
+            te
+          </mark>
+          <span
+            class="ais-Highlight-nonHighlighted"
+          >
+            st
+          </span>
+        </span>
+      </div>
+    `);
+  });
+
+  test("renders nothing when there's no match", () => {
+    const { container } = render(
+      <Highlight
+        hit={{
+          objectID: '1',
+          __position: 1,
+        }}
+        // @ts-expect-error TS doesn't allow an attribute which doesn't exist
+        attribute="something-that-doesnt-exist"
+      />
+    );
+
+    expect(container).toMatchInlineSnapshot(`
+      <div>
+        <span
+          class="ais-Highlight"
+        />
+      </div>
+    `);
+  });
+
+  test('forwards className and root props', () => {
+    const { container } = render(
+      <Highlight
+        className="custom-rootclass"
+        hidden={true}
+        hit={{
+          objectID: '1',
+          __position: 1,
+        }}
+        // @ts-expect-error TS doesn't allow an attribute which doesn't exist
+        attribute="something-that-doesnt-exist"
```

Comment:
We don't need this in that test.

```suggestion
```

Reply:
We can pass a valid attribute then, so we don't need the `@ts-expect-error`. This isn't what we're testing here, and it distracts from the actual test case.